### PR TITLE
add a close button to the main toolbar when running on gamescope

### LINKED ISSUE #522: Add a close button to the toolbar in the main GUI
### Role

Running Prism Launcher on SteamOS (in Wayland instance with no window decorations)

### Suggestion

Add a close button to the toolbar in the main GUI

### Benefit

So I can more easily close Prism Launcher without killing it outside the app

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

On SteamOS, when running PrismLauncher from Game Mode, it is run in an instance of wayland that does not include window decorations, which means that there's no easy way to close it when running on the Steam Deck. The "File" menu also does not include a Close or Exit option, meaning that the only way to exit Prism Launcher is by returning to the Steam Deck game mode and selecting "Exit Prism Launcher", which sends a sigterm.

### PR DESCRIPTION
**Pull request by leia-uwu**
i don't own a steam deck to test this, but it worked on a local gamescope session so ig it will also work on the steam deck

closes #522

### CODE REVIEW

**Comment by leia-uwu** [comment]
(forgot to sign the commit lol)

**Comment by Scrumplex** [comment]
I wonder if there is a more general solution. Perhaps we can somehow query if server-side decorations are present and add this action depending on that?

**Comment by leia-uwu** [comment]
> I wonder if there is a more general solution. Perhaps we can somehow query if server-side decorations are present and add this action depending on that?

hmm, gamescope only supports xwayland clients so idk :ioa:
when running natively on wayland qt adds client side decorations if theres no server side ones

**Comment by leia-uwu** [comment]
~~i totally didn't do this pr while sleep deprived and wrote instance toolbar instead of main toolbar lmao~~

**Comment by Ryex** [comment]
It actualy would be nice to detect the case of no window decorations to add the close button. I use tileing windows managers like bspwm without decorations from time to time, that would be a nice to have.

**Review by Scrumplex — APPROVED** [review]
Maybe just add a FIXME comment to make this more universal in the future

**Review by DioEgizio — APPROVED** [review]

File: launcher/ui/MainWindow.cpp [hunk 1]
```diff
@@ -219,6 +219,12 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new Ui::MainWi
         // disabled until we have an instance selected
         ui->instanceToolBar->setEnabled(false);
         setInstanceActionsEnabled(false);
+
+        // add a close button at the end of the main toolbar when running on gamescope / steam deck
```
Comment (Scrumplex) [inline launcher/ui/MainWindow.cpp:223]:
```suggestion
        // add a close button at the end of the main toolbar when running on gamescope / steam deck
        // FIXME: detect if we don't have server side decorations instead
```

**Pull request merged**